pkg/cloud/secret.go

```diff
@@ -17,34 +17,31 @@ package cloud
 
 import (
 	"context"
-	"encoding/json"
+	"errors"
 	"fmt"
-	"hash/crc32"
 
-	gsm "cloud.google.com/go/secretmanager/apiv1"
-	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
 	"github.com/go-logr/logr"
 	vapi "github.com/vertica/vertica-kubernetes/api/v1"
 	"github.com/vertica/vertica-kubernetes/pkg/events"
+	"github.com/vertica/vertica-kubernetes/pkg/secrets"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// MultiSourceSecretFetcher is secret reader that handles retrival from
-// different sources such as Kubernetes secret store and Google Secrets Manager
-// (GSM).
-type MultiSourceSecretFetcher struct {
+// ControllerSecretFetcher is secret reader designed for the verticadb
+// controller. It can handle retrival from different sources, such as Kubernetes
+// secret store, Google Secrets Manager (GSM), etc.
+type ControllerSecretFetcher struct {
```

Comment:
Isn't `VerticaDBSecretFetcher` more accurate?

Reply:
I like this name.